### PR TITLE
Feature/xml parser, fixes #27

### LINKED ISSUE #27: Create XML level parser
Create a method which can parse an XML file into a level. Player, Balls and background should be parsed

### PR DESCRIPTION
**Pull request by TMRert**
implemented the xml parser, levels can now be written in XML format and be parsed into actual playable levels

### CODE REVIEW

File: src/main/java/net/liquidpineapple/pang/objects/GameObject.java [hunk 1]
```diff
@@ -22,18 +22,21 @@
...
+    private int objectHeight;
     private Image image;
 
     public GameObject(String textureLocation, int startX, int startY) {
+        this.xPos = startX;
+        this.yPos = startY;
+        ChangeImage(textureLocation);
+    }
+
+    public void ChangeImage(String textureLocation){
```
Comment (Fastjur) [inline src/main/java/net/liquidpineapple/pang/objects/GameObject.java]: By convention methods are camelCased, not PascalCased ;) @TMRert

File: src/main/java/net/liquidpineapple/pang/objects/Ball.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+        if(sizeIn>0 && sizeIn<5){
+        size = sizeIn;}
+        else {
+            size = 4;}
+
+        //Makes the ball go either left or right depending on input, if the input is incorrect it randomly chooses a direction.
+        if(direction.equals(Directions.LEFT)){
+            movX=-15;}
+        else if(direction.equals(Directions.RIGHT)){
```
Comment (Fastjur) [inline src/main/java/net/liquidpineapple/pang/objects/Ball.java]: @Wolgo Please adhere to the styling conventions here. The `} else` not being on a separate line drastically reduces readability

File: src/main/java/net/liquidpineapple/pang/objects/HookAndRope.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        if(this.inUse) {
+            move();
+        }
+    }
+
+    /**
+     * Get method to get the boolean inUse, indicating if the hook is in use.
+     * @return boolean inUse;
+     */
+    public boolean isInUse() {
```
Comment (Fastjur) [inline src/main/java/net/liquidpineapple/pang/objects/HookAndRope.java]: I have replaced this with the `@Getter` lombok annotation te remove annoying clutter :D

File: src/main/java/net/liquidpineapple/pang/screens/Level.java [hunk 4]
```diff
@@ -16,24 +25,111 @@
...
-        );
+                for (Ball ball : loadBalls(doc.getElementsByTagName("ball"))) {
+                    output.objectList.add(ball);
+                }
+
+                for (Player player : loadPlayer(doc.getElementsByTagName("player"))) {
+                    output.objectList.add(player);
+                }
+            }
+            catch (Exception e) {
```
Comment (Fastjur) [inline src/main/java/net/liquidpineapple/pang/screens/Level.java]: Do we really want to catch ALL exceptions here?

File: src/main/java/net/liquidpineapple/pang/screens/Level.java [hunk 5]
```diff
@@ -16,24 +25,111 @@
...
+    public static ArrayList<Ball> loadBalls(NodeList balllist){
+        ArrayList<Ball> ballArray = new ArrayList<Ball>();
+        for(int i =0; i<balllist.getLength(); i++){
+            Node ballNode = balllist.item(i);
+            Element ballElement = (Element) ballNode;
+            int int_Xpos = Integer.parseInt(ballElement.getElementsByTagName("x").item(0).getTextContent());
+            int int_Ypos = Integer.parseInt(ballElement.getElementsByTagName("y").item(0).getTextContent());
+            String direction = ballElement.getElementsByTagName("direction").item(0).getTextContent();
+            int size = Integer.parseInt(ballElement.getElementsByTagName("size").item(0).getTextContent());
+            String color = ballElement.getElementsByTagName("color").item(0).getTextContent();
```
Comment (Fastjur) [inline src/main/java/net/liquidpineapple/pang/screens/Level.java]: Color is determined solely by size right? Do we need this? @Wolgo @TMRert

Comment (grappigegovert) [inline src/main/java/net/liquidpineapple/pang/screens/Level.java]: I'd say color and size are two separate properties

**Comment by Fastjur** [comment]
@Wolgo @TMRert Please look at the changes I made to familiarize yourself with the slightly changed code.

Merging!

**Pull request merged**